core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -330,13 +333,11 @@ class PartKeyLuceneIndex(dataset: Dataset,
       .withTag("shard", shardNum)
       .start()
     val collector = new PartIdStartTimeCollector()
-    partIds.grouped(512).foreach { batch => // limit on query clause count is 1024, hence batch
-      val booleanQuery = new BooleanQuery.Builder
-      batch.foreach { pId =>
-        booleanQuery.add(new TermQuery(new Term(PART_ID, pId.toString)), Occur.SHOULD)
-      }
-      searcherManager.acquire().search(booleanQuery.build(), collector)
+    val terms = new util.ArrayList[BytesRef]()
+    partIds.foreach { pId =>
+      terms.add(new BytesRef(pId.toString.getBytes))
```

Comment:
Is there a way to not need a huge amount of heap for each Int (4 bytes)?
This requires, per partition, an Array[Byte], which is >= 28 bytes, plus another BytesRef which is another 28+ bytes.  Wowzas.

Reply:
Also, is it safe to remove the grouping of 512 terms?  How is this code not requiring the previous grouping?

Reply:
(please add docs to explain why it's OK)

Reply:
Evan, unfortunately we are limited by the capabilities Lucene offers. The TermInSetQuery requires BytesRef.

I have confirmed that TermInSetQuery does not limit number of terms. The unit  test confirms that it can exceed the 1024 limit of BooleanQuery.

I will add docs.